In [2]:
import pandas as pd
import numpy as np

In [3]:
path = 'poblacion_global_2020.csv'

# Ejercicio

Examina detenidamente el siguiente dataset y responde las siguientes preguntas:

* ¿Porqué cumple con las condiciones de Tidy Data?
* ¿Cuál es el data point principal y cual es la meta data?
* Si quisieramos saber más sobre los niveles educativos globales ¿Qué otras características necesitariamos para este data set?
* ¿Qué tipo de análisis harías con este data set?

In [4]:
df = pd.read_csv(path)

df.head(10)

Pais  Poblacion (2020)  Densidad  Territorio (Km²)  \
0           China        1440297825       153           9388211   
1           India        1382345085       464           2973190   
2  Estados Unidos         331341050        36           9147420   
3       Indonesia         274021604       151           1811570   
4        Pakistán         221612785       287            770880   
5          Brasil         212821986        25           8358140   
6         Nigeria         206984347       226            910770   
7      Bangladesh         164972348      1265            130170   
8           Rusia         145945524         9          16376870   
9          Mexico         129166028        66           1943950   

   Tasa Migracion Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
0       -348399.0             1.7            38               61 %  
1       -532687.0             2.2            28               35 %  
2        954806.0             1.8            38               83 %  
3        -98955.0             2.3            30               56 %  
4       -233379.0             3.6            23               35 %  
5         21200.0             1.7            33               88 %  
6        -60000.0             5.4            18               52 %  
7       -369501.0             2.1            28               39 %  
8        182456.0             1.8            40               74 %  
9        -60000.0             2.1            29               84 %

**¿Porqué cumple con las condiciones de Tidy Data?**

Se cumplen las condiciones del Tidy Data ya que los datos están estructurados de forma que cada fila es una observación y cada columna una característica de dicha observación


**¿Cuál es el data point principal y cual es la meta data?**

En este caso el data principal es la población de cada pais y el meta data es el resto de caracteristicas (indicadas en cada columna) asociadas a cada pais, es decir, densidad, extensión del territorio, Tasa de Migración, etc.

**Si quisieramos saber más sobre los niveles educativos globales ¿Qué otras características necesitariamos para este data set?**

Para poder saber los niveles educativos globales se deberían indicar caracteristicas que reflejen el nivel educativo de la población. En este caso, establecer unas características globales es complicado por las diferentes características intrinsecas de cada país con sistemas educativos muy diferentes entre ellos y difícilmente comparables. Sin embargo, como aproximación se podrían establecer las siguientes características bien en términos absolutos de población o porcentuales respecto a la población global de cada país:
- Población sin nivel educativo (Población que no tiene ningún tipo de eduación reglada)
- población con nivel educativo básico (Población que ha completado el nivel básico de educación)
- Población con nivel educativo medio (Población que formación profesional, o equiparable, de algún tipo)
- Población con nivel educativo superior (Población con formación universitaria o equiparable)

**¿Qué tipo de análisis harías con este data set?**

Se puede analizar:
- la evolución de la población por paises en el año en cuestión con las características Tasa de migración y Tasa de Fertilidad que nos indican el flujo de población entrante y saliente del pais.
- La distribucion de la población. Con la característica de la densidad de la población se sabe la cantidad de personas por unidad de superficie por lo que se puede saber si hay mucha o poca población en fucnión de la extensión del pais. Además, el campo %población urbana nos indica si la población se congrega más en las ciudades o en las zonas rurales lo que nos puede dar una idea del nivel de industrialización de los paises en cuestión
- La piramide de edad de los paises. En lugares con tasa de fertilidad muy alta y promedio de edad bajo se puede deducir que la cantidad de población joven es mayor por lo que la piramide de edad tendrá una base más amplia y viceversa

# Ejercicios:

Manipula el dataset, de manera que respondas las siguientes preguntas:

1. ¿Cual es el promedio de Tasa de Fertilidad global?
2. ¿Que país tiene la Edad Promedio más alta y cual la más baja?
3. Agrupa los países por continente y obtén la mediana de la población urbana
4. ¿Qué otro de cálculo de interés (usando estas variables) crees pertinente para comparar la calidad de vida a nivel global? ¿Porqué?

In [9]:
data_filter = df[df['Pais'] == 'España']#prueba de filtrado
data_filter

Pais  Poblacion (2020)  Densidad  Territorio (Km²)  Tasa Migracion  \
29  España          46757980        94            498800         40000.0   

   Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
29             1.3            45               80 %

Compruebo los tipos de datos que hay en la tabla

In [5]:
df.dtypes


Pais                   object
Poblacion (2020)        int64
Densidad                int64
Territorio (Km²)        int64
Tasa Migracion        float64
Tasa Fertilidad        object
Edad Promedio          object
% Poblacion Urbaba     object
dtype: object

**¿Cual es el promedio de Tasa de Fertilidad global?**

La Tasa de fertilidad es de tipo objeto por lo que no me permite realizar operaciones con él. Je intentado convertilo a tipo float pero no me ha dejado inidcandome que hay registros donde los valores no son convertibles por figurar N.A.
A continuación detecto los registros con valor N.A.

In [5]:
data_NA = df[df['Tasa Fertilidad'] == 'N.A.']#compruebo los registros sin un dato númerico
data_NA

Pais  Poblacion (2020)  Densidad  Territorio (Km²)  \
201              Isla de Man             85112       149               570   
202                  Andorra             77287       164               470   
203                 Dominica             72017        96               750   
204             Islas Cayman             65854       274               240   
205                  Bermuda             62237      1246                50   
206           Islas Marshall             59259       329               180   
207  Islas Mariana del Norte             57619       125               460   
208              Groenlandia             56787         0            410450   
209          Samoa Americana             55169       276               200   
210      Saint Kitts y Nevis             53264       205               260   
211              Islas Feroe             48896        35              1396   
212               San Martín             42960      1261                34   
213                   Monaco             39290     26337                 1   

     Tasa Migracion Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
201             NaN            N.A.          N.A.               53 %  
202             NaN            N.A.          N.A.               88 %  
203             NaN            N.A.          N.A.               74 %  
204             NaN            N.A.          N.A.               97 %  
205             NaN            N.A.          N.A.               97 %  
206             NaN            N.A.          N.A.               70 %  
207             NaN            N.A.          N.A.               88 %  
208             NaN            N.A.          N.A.               87 %  
209             NaN            N.A.          N.A.               88 %  
210             NaN            N.A.          N.A.               33 %  
211             NaN            N.A.          N.A.               43 %  
212             NaN            N.A.          N.A.               96 %  
213             NaN            N.A.          N.A.               N.A.

Del registro con índice 201 al 213 tienen valor N.A. en la carcaterística Tasa de Fertilidad.
A continuación ordeno el dataframe tanto de menor a mayor como de mayor a menor para ver si hay algún registro vacio o con cualquier otro valor que no sea numérico para no tenerlo en cuenta en el calculo de la media

In [6]:
orden_fertilidad = df.sort_values('Tasa Fertilidad', ascending=False)
orden_fertilidad.head(30)

Pais  Poblacion (2020)  Densidad  Territorio (Km²)  \
213                    Monaco             39290     26337                 1   
208               Groenlandia             56787         0            410450   
201               Isla de Man             85112       149               570   
202                   Andorra             77287       164               470   
203                  Dominica             72017        96               750   
205                   Bermuda             62237      1246                50   
206            Islas Marshall             59259       329               180   
207   Islas Mariana del Norte             57619       125               460   
204              Islas Cayman             65854       274               240   
209           Samoa Americana             55169       276               200   
211               Islas Feroe             48896        35              1396   
210       Saint Kitts y Nevis             53264       205               260   
212                San Martín             42960      1261                34   
55                      Niger          24346468        19           1266700   
72                    Somalia          15965848        25            627340   
15                   DR Congo          90003954        40           2267050   
59                       Mali          20346106        17           1220190   
71                       Chad          16502877        13           1259200   
43                     Angola          33032075        26           1246700   
77                    Burundi          11948454       463             25680   
6                     Nigeria         206984347       226            910770   
143                    Gambia           2427782       239             10120   
58               Burkina Faso          20997293        76            273600   
30                     Uganda          45974931       229            199810   
23                   Tanzania          60012400        67            885800   
76                      Benin          12175480       108            112760   
45                 Mozambique          31398811        40            786380   
124  República Centroafricana           4843954         8            622980   
64                     Zambia          18468257        25            743390   
69                    Senegal          16816539        87            192530   

     Tasa Migracion Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
213             NaN            N.A.          N.A.               N.A.  
208             NaN            N.A.          N.A.               87 %  
201             NaN            N.A.          N.A.               53 %  
202             NaN            N.A.          N.A.               88 %  
203             NaN            N.A.          N.A.               74 %  
205             NaN            N.A.          N.A.               97 %  
206             NaN            N.A.          N.A.               70 %  
207             NaN            N.A.          N.A.               88 %  
204             NaN            N.A.          N.A.               97 %  
209             NaN            N.A.          N.A.               88 %  
211             NaN            N.A.          N.A.               43 %  
210             NaN            N.A.          N.A.               33 %  
212             NaN            N.A.          N.A.               96 %  
55           4000.0               7            15               17 %  
72         -40000.0             6.1            17               47 %  
15          23861.0               6            17               46 %  
59         -40000.0             5.9            16               44 %  
71           2000.0             5.8            17               23 %  
43           6413.0             5.6            17               67 %  
77           2001.0             5.5            17               14 %  
6          -60000.0             5.4            18               52 %  
143         -

In [7]:
orden_fertilidad_menor = df.sort_values('Tasa Fertilidad')
orden_fertilidad_menor.head(30)

Pais  Poblacion (2020)  Densidad  Territorio (Km²)  \
27            Corea del Sur          51276977       527             97230   
56                   Taiwan          23824369       673             35410   
113               Singapore           5858322      8358               700   
166                   Macao            650846     21645                30   
140             Puerto Rico           2846860       323              8870   
29                   España          46757980        94            498800   
22                   Italia          60446035       206            294140   
157                  Chipre           1208886       131              9240   
130                Moldavia           4032294       123             32850   
103               Hong Kong           7507523      7140              1050   
134    Bosnia y Herzegovina           3277152        64             51000   
86                   Grecia          10413936        81            128900   
88                 Portugal          10191409       111             91590   
37                  Polonia          37839255       124            306230   
188             Santa Lucia            183774       301               610   
34                  Ucrania          43686577        75            579320   
129                 Croacia           4100719        73             55960   
92   Emiratos Árabes Unidos           9910892       118             83600   
10                    Japan         126407422       347            364555   
156                Mauricio           1272140       626              2030   
115               Finlandia           5542237        18            303890   
117              Eslovaquia           5460109       114             48088   
100                   Suiza           8665615       219             39516   
189             Islas Canal            174140       915               190   
168              Luxemburgo            627704       242              2590   
19                Tailandia          69830779       137            510890   
93                  Hungría           9655983       107             90530   
98                   Serbia           8731081       100             87460   
38                   Canada          37799407         4           9093510   
147     Macedonia del Norte           2083359        83             25220   

     Tasa Migracion Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
27          11731.0             1.1            44               82 %  
56          30001.0             1.2            42               79 %  
113         27028.0             1.2            42               N.A.  
166          5000.0             1.2            39               N.A.  
140        -97986.0             1.2            44               N.A.  
29          40000.0             1.3            45               80 %  
22         148943.0             1.3            47               69 %  
157          5000.0             1.3            37               67 %  
130         -1387.0             1.3            38               43 %  
103         29308.0             1.3            45               N.A.  
134        -21585.0             1.3            43               52 %  
86         -16000.0             1.3            46               85 %  
88          -6000.0             1.3            46               66 %  
37         -29395.0             1.4            42               60 %  
188             0.0             1.4            34               19 %  
34          10000.0             1.4            41               69 %  
129         -8001.0             1.4            44               58 %  
92          40000.0             1.4            33               86 %  
10          71560.0             1.4            48               92 %  
156             0.0             1.4            37               41 %  
115         14000.0             1.5            43               86 %  
117          1485.0             1.5            41               54 %  
10

Calculo la media entre los registros con un valor númerico en la característica Tasa de Fertilidad ya para ello:
- Convierto todos los valores de la Tasa de Fertilidad con vaores númericos (indices de 0 a 200) de Objeto a float
- Una vez convertidos a tipo float calculo la media

In [5]:
df[0:201]['Tasa Fertilidad'].astype('float64').mean()#calculo la media de los datos con valor númerico tras convertirlos en float


2.6930348258706465

**¿Que país tiene la Edad Promedio más alta y cual la más baja?**

Edad Promedio es de tipo objeto por lo que hay que transformarlo a un tipo numerico pero para ello primero hay que comprobar los registros con Edad Promedio que no tengan un valor numerico, en este caso N.A.

In [13]:
df[df['Edad Promedio']=='N.A.']

Pais  Poblacion (2020)  Densidad  Territorio (Km²)  \
201              Isla de Man             85112       149               570   
202                  Andorra             77287       164               470   
203                 Dominica             72017        96               750   
204             Islas Cayman             65854       274               240   
205                  Bermuda             62237      1246                50   
206           Islas Marshall             59259       329               180   
207  Islas Mariana del Norte             57619       125               460   
208              Groenlandia             56787         0            410450   
209          Samoa Americana             55169       276               200   
210      Saint Kitts y Nevis             53264       205               260   
211              Islas Feroe             48896        35              1396   
212               San Martín             42960      1261                34   
213                   Monaco             39290     26337                 1   

     Tasa Migracion Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
201             NaN            N.A.          N.A.               53 %  
202             NaN            N.A.          N.A.               88 %  
203             NaN            N.A.          N.A.               74 %  
204             NaN            N.A.          N.A.               97 %  
205             NaN            N.A.          N.A.               97 %  
206             NaN            N.A.          N.A.               70 %  
207             NaN            N.A.          N.A.               88 %  
208             NaN            N.A.          N.A.               87 %  
209             NaN            N.A.          N.A.               88 %  
210             NaN            N.A.          N.A.               33 %  
211             NaN            N.A.          N.A.               43 %  
212             NaN            N.A.          N.A.               96 %  
213             NaN            N.A.          N.A.               N.A.

Estamos en el mismo caso que antes.

In [35]:
df[df['Edad Promedio']==df[0:201]['Edad Promedio'].max()]#obtengo todos los datos del registro


Pais  Poblacion (2020)  Densidad  Territorio (Km²)  Tasa Migracion  \
10  Japan         126407422       347            364555         71560.0   

   Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
10             1.4            48               92 %

Si sólo quiero el nombre del pais obtengo el índice para obtener el pais

In [50]:
indice=df.index[df['Edad Promedio'] == df[0:201]['Edad Promedio'].max()].tolist()#obtengo el índice del pais en forma de lista

In [54]:
df['Pais'][indice]

10    Japan
Name: Pais, dtype: object

In [33]:
df[df['Edad Promedio']==df[0:201]['Edad Promedio'].min()]#obtengo el minimo

Pais  Poblacion (2020)  Densidad  Territorio (Km²)  Tasa Migracion  \
55  Niger          24346468        19           1266700          4000.0   

   Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
55               7            15               17 %

**Agrupa los países por continente y obtén la mediana de la población urbana**


In [60]:
df[df['% Poblacion Urbaba']=='N.A.']

Pais  Poblacion (2020)  Densidad  Territorio (Km²)  \
49     Venezuela          28421581        32            882050   
103    Hong Kong           7507523      7140              1050   
113    Singapore           5858322      8358               700   
128       Kuwait           4281320       240             17820   
140  Puerto Rico           2846860       323              8870   
166        Macao            650846     21645                30   
175   Guadeloupe            400136       237              1690   
213       Monaco             39290     26337                 1   

     Tasa Migracion Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
49        -653249.0             2.3            30               N.A.  
103         29308.0             1.3            45               N.A.  
113         27028.0             1.2            42               N.A.  
128         39520.0             2.1            37               N.A.  
140        -97986.0             1.2            44               N.A.  
166          5000.0             1.2            39               N.A.  
175         -1440.0             2.2            44               N.A.  
213             NaN            N.A.          N.A.               N.A.

In [9]:
urbana_not_na = df[df['% Poblacion Urbaba'] != 'N.A.']#filtro el dataframe para que sólo me salgan los registros con valores porcentuales
urbana_not_na

Pais  Poblacion (2020)  Densidad  Territorio (Km²)  \
0                  China        1440297825       153           9388211   
1                  India        1382345085       464           2973190   
2         Estados Unidos         331341050        36           9147420   
3              Indonesia         274021604       151           1811570   
4               Pakistán         221612785       287            770880   
..                   ...               ...       ...               ...   
208          Groenlandia             56787         0            410450   
209      Samoa Americana             55169       276               200   
210  Saint Kitts y Nevis             53264       205               260   
211          Islas Feroe             48896        35              1396   
212           San Martín             42960      1261                34   

     Tasa Migracion Tasa Fertilidad Edad Promedio % Poblacion Urbaba  
0         -348399.0             1.7            38               61 %  
1         -532687.0             2.2            28               35 %  
2          954806.0             1.8            38               83 %  
3          -98955.0             2.3            30               56 %  
4         -233379.0             3.6            23               35 %  
..              ...             ...           ...                ...  
208             NaN            N.A.          N.A.               87 %  
209             NaN            N.A.          N.A.               88 %  
210             NaN            N.A.          N.A.               33 %  
211             NaN            N.A.          N.A.               43 %  
212             NaN            N.A.          N.A.               96 %  

[206 rows x 8 columns]

In [10]:
#convierto los valores de tipo objeto a float y para ello elimino los simbolos % a la derecha
mediana_urb = urbana_not_na['% Poblacion Urbaba'].str.rstrip('%').astype('float').median()

print('La mediana global de la población urbana es ', mediana_urb, '%')

La mediana global de la población urbana es  61.5 %


Para agruparlo por continentes primero obtengo un .csv donde están los paises con el continente al que pertenecen

In [11]:
ruta='paises_estandar.csv'#obtengo un csv con los paises y continente al que pertenecen

In [12]:
paises = pd.read_csv(ruta, sep=';') #genero el dataframe correspondiente teniendo en cuenta que el separador es ;
paises.head(10)

Continente       Pais
0     EUROPA    Austria
1     EUROPA    Bélgica
2     EUROPA   Bulgaria
3     EUROPA     Chipre
4     EUROPA  Dinamarca
5     EUROPA     España
6     EUROPA  Finlandia
7     EUROPA    Francia
8     EUROPA     Grecia
9     EUROPA    Hungría

In [13]:
paises.dtypes

Continente    object
Pais          object
dtype: object

In [14]:
# hago una unión entre los 2 dataframe (df y paises) por el campo común Pais.
continentes_add_pob_urb = pd.merge(paises,urbana_not_na, how='right')

continentes_add_pob_urb

Continente                 Pais  Poblacion (2020)  Densidad  \
0         ASIA                China        1440297825       153   
1         ASIA                India        1382345085       464   
2      AMERICA       Estados Unidos         331341050        36   
3         ASIA            Indonesia         274021604       151   
4         ASIA             Pakistán         221612785       287   
..         ...                  ...               ...       ...   
201    AMERICA          Groenlandia             56787         0   
202    OCEANIA      Samoa Americana             55169       276   
203    AMERICA  Saint Kitts y Nevis             53264       205   
204     EUROPA          Islas Feroe             48896        35   
205    AMERICA           San Martín             42960      1261   

     Territorio (Km²)  Tasa Migracion Tasa Fertilidad Edad Promedio  \
0             9388211       -348399.0             1.7            38   
1             2973190       -532687.0             2.2            28   
2             9147420        954806.0             1.8            38   
3             1811570        -98955.0             2.3            30   
4              770880       -233379.0             3.6            23   
..                ...             ...             ...           ...   
201            410450             NaN            N.A.          N.A.   
202               200             NaN            N.A.          N.A.   
203               260             NaN            N.A.          N.A.   
204              1396             NaN            N.A.          N.A.   
205                34             NaN            N.A.          N.A.   

    % Poblacion Urbaba  
0                 61 %  
1                 35 %  
2                 83 %  
3                 56 %  
4                 35 %  
..                 ...  
201               87 %  
202               88 %  
203               33 %  
204               43 %  
205               96 %  

[206 rows x 9 columns]

El atributo how es para que  muestre todos los registros del dataframe que me interesa que es df aunque no tengan correspondencia en paises.

In [15]:
continentes_add_pob_urb[continentes_add_pob_urb.Continente.isnull()] #compruebo que no haya ningún registro con la caracteristica Continente vacia

Empty DataFrame
Columns: [Continente, Pais, Poblacion (2020), Densidad, Territorio (Km²), Tasa Migracion, Tasa Fertilidad, Edad Promedio, % Poblacion Urbaba]
Index: []

In [16]:
continentes_add_pob_urb['% Poblacion Urbaba']=continentes_add_pob_urb['% Poblacion Urbaba'].str.rstrip('%').astype('float64')#Convierto tipo objeto en tipo float para hacer calculos
continentes_add_pob_urb.dtypes

Continente             object
Pais                   object
Poblacion (2020)        int64
Densidad                int64
Territorio (Km²)        int64
Tasa Migracion        float64
Tasa Fertilidad        object
Edad Promedio          object
% Poblacion Urbaba    float64
dtype: object

In [17]:
continentes_add_pob_urb.groupby('Continente').median()['% Poblacion Urbaba']#obtengo la mediana por continente

Continente
AFRICA     46.0
AMERICA    74.0
ASIA       60.0
EUROPA     69.0
OCEANIA    45.0
Name: % Poblacion Urbaba, dtype: float64

**¿Qué otro de cálculo de interés (usando estas variables) crees pertinente para comparar la calidad de vida a nivel global? ¿Porqué?**

Las características de lo datos dados que considero más importantes para poder determinar la calidad de vida son:
- Tasa de Migración: en los paises donde hay una mayor calidad de vida las tasas tienden a ser positivas. Sin embargo, hay factores como el contexto de la zona (guerras, desastres naturales, ubicación geografica del Pais, etc.) que puede influir en esta variable por lo que los resultados se deben contrastar con otras variables.
- Edad Promedio: en los paises con mayor calidad de vida los promedios de edad son mayores ya que la población vive más por tener unas mejores condiciones de vida. Está característica está afetada por la Tasa de fertilidad.
- Tasa de Fertilidad: en los paises con un mayor nivel de desarrollo la tasa de fertilidad es más baja. Esto se debe a varios factores como por ejemplo:
    - Menos muertes en la infancia, lo que significa que las mujeres tienen menos bebés.
    - Mayor acceso a la anticoncepción.
    - Más mujeres en la educación y el trabajo.
- Porcentaje de población en medio urbano: Está característica suele estar ligada a un mayor desrrollo del País a nivel industrial y tecnológico

Respecto a los cálculos ha realizar para poder hacernos una idea de la calidad de vida a nivel global es importante calcular al menos los siguientes:
- Media y mediana para ver entorno a que valores se agrupan los datos
- desviación tipica para conocer el grado de dispersión de los valores respecto a la media
- cuartiles, rango: es importante conocer como se localizan los diferentes registros que forman parte del dataset en función del valor de la característica que nos interese.

A continuación se realiza un calculo estadistico tanto a nivel global como por continentes de las variables indicadas anteriormente. El calculo global nos sirve de comparativa respecto a lo que obtenemos por continente. 

### Estadisticas Tasa de Migración

#### Globales

In [18]:
df.describe()['Tasa Migracion']

count       201.000000
mean          6.283582
std      123291.887548
min     -653249.000000
25%      -10047.000000
50%        -852.000000
75%        9741.000000
max      954806.000000
Name: Tasa Migracion, dtype: float64

#### Por continente

In [19]:
continentes_add_df = pd.merge(paises,df, how='right')
continentes_add_df.groupby('Continente').describe()['Tasa Migracion']

count          mean            std       min      25%     50%  \
Continente                                                                  
AFRICA       57.0  -8123.684211   44731.829430 -174200.0 -10047.0 -4000.0   
AMERICA      40.0  16858.600000  194150.526146 -653249.0  -9961.0  -325.5   
ASIA         47.0 -42514.063830  132484.566852 -532687.0 -42506.0  -852.0   
EUROPA       44.0  37144.795455  103935.387549  -73999.0  -4849.5  4500.0   
OCEANIA      13.0  11827.615385   44277.594602   -6202.0  -1600.0  -800.0   

                 75%       max  
Continente                      
AFRICA          0.00  168694.0  
AMERICA      1950.00  954806.0  
ASIA        23236.00  134979.0  
EUROPA      30131.75  543822.0  
OCEANIA       120.00  158246.0

A nivel de continente se observa que la media de de la Tasa migración de los paises de Europa, America y Oceania es positiva mientras que en Asia y Africa es negativa. Sin embargo, la desviación tipica es elevada en todos los continentes lo que da una idea de los valores tan diferentes que existen entre los paises. Si observamos los cuartiles se ve que al menos el 50% de los paises en todos los continentes excepto en Europa presentan Tasas de migración negativa siendo del 75 % de los paises en el caso de Africa.

### Estadisticas Edad Promedio

#### Globales

In [20]:
edad_promedio_not_na = df[df['Edad Promedio']!='N.A.']
edad_promedio = edad_promedio_not_na['Edad Promedio'].astype('float').describe()
edad_promedio

count    201.000000
mean      30.606965
std        9.127965
min       15.000000
25%       22.000000
50%       30.000000
75%       39.000000
max       48.000000
Name: Edad Promedio, dtype: float64

#### Por continente

In [21]:
continentes_add_edad_promedio = pd.merge(paises,edad_promedio_not_na, how='right')
continentes_add_edad_promedio['Edad Promedio']=continentes_add_edad_promedio ['Edad Promedio'].astype('float64')
continentes_add_edad_promedio.groupby('Continente').describe()['Edad Promedio']

count       mean       std   min   25%   50%   75%   max
Continente                                                          
AFRICA       57.0  21.456140  5.203262  15.0  18.0  19.0  23.0  37.0
AMERICA      40.0  32.825000  6.480296  23.0  28.0  32.0  36.5  47.0
ASIA         47.0  30.680851  6.750544  18.0  26.0  30.0  32.5  48.0
EUROPA       44.0  41.522727  3.202387  32.0  40.0  42.0  43.0  47.0
OCEANIA      13.0  26.692308  6.835991  20.0  22.0  23.0  34.0  38.0

Respecto a la edad promedio se observa como es en Europa donde la edad es mayor y con una deviación típica bastante pequeña lo que habla de la uniformidad de los valores en todos los paises que se analizan en este continente, lo que se confirma si observamos los cuartiles. Respecto al resto de continentes destaca Africa con la edad promedio media menor de todos los continentes y con un 75 % de los paises con una edad promedio menor de 23 años. Oceanía también presenta una media de la edad promedio baja con el 50 % de los paises por debajo de 23 años de edad promedio aunque hay que valorar que la cantidad de paises analizados es muy baja con repecto a los otros continentes. Africa y Oceanía son los únicos continentes que se sitúan por debajo de los valores obtenidos globalmente.

### Estadisticas Tasa de Fertilidad

#### Globales

In [22]:
tasa_fertil_not_na = df[df['Tasa Fertilidad'] != 'N.A.']
tasa_fertil=tasa_fertil_not_na['Tasa Fertilidad'].astype('float').describe()
tasa_fertil

count    201.000000
mean       2.693035
std        1.264496
min        1.100000
25%        1.700000
50%        2.300000
75%        3.600000
max        7.000000
Name: Tasa Fertilidad, dtype: float64

#### Por continente

In [23]:
continentes_add_tasa_ferti = pd.merge(paises,tasa_fertil_not_na, how='right')
continentes_add_tasa_ferti['Tasa Fertilidad']=continentes_add_tasa_ferti ['Tasa Fertilidad'].astype('float64')
continentes_add_tasa_ferti.groupby('Continente').describe()['Tasa Fertilidad']

count      mean       std  min  25%   50%    75%  max
Continente                                                       
AFRICA       57.0  4.143860  1.173520  1.4  3.3  4.40  4.800  7.0
AMERICA      40.0  2.110000  0.452231  1.2  1.8  2.05  2.400  3.4
ASIA         47.0  2.336170  0.778355  1.1  1.9  2.20  2.800  4.6
EUROPA       44.0  1.597727  0.208526  1.3  1.5  1.60  1.725  2.1
OCEANIA      13.0  3.123077  0.922093  1.8  2.0  3.60  3.800  4.4

En la tasa de Fertilidad destacan tanto Africa como Oceanía con valores por encima de la media global y por debajo se sitúa Europa con unos valores bajos y un rango también muy bajo lo que vuelve a dejar constancia de la unifomridad en cuanto a los valores obtenidos para esta variable en este continente. En Africa el rango es elevado lo que indica una difernecia notable entre paises. En Oceanía aunque la diferencia también es importante es bastante más baja que en Africa

### Estadisticas % Población Urbana

#### Globales

In [25]:
continentes_add_pob_urb.describe()['% Poblacion Urbaba']

count    206.000000
mean      60.368932
std       22.661880
min       13.000000
25%       43.000000
50%       61.500000
75%       79.750000
max      100.000000
Name: % Poblacion Urbaba, dtype: float64

#### Por continente

In [27]:
continentes_add_pob_urb.groupby('Continente').describe()['% Poblacion Urbaba']

count       mean        std   min    25%   50%   75%    max
Continente                                                             
AFRICA       57.0  49.280702  19.792454  14.0  37.00  46.0  64.0  100.0
AMERICA      43.0  68.767442  22.266607  19.0  54.00  74.0  86.5   97.0
ASIA         45.0  60.066667  23.779670  18.0  38.00  60.0  80.0   96.0
EUROPA       47.0  70.170213  15.040986  30.0  58.50  69.0  82.5   98.0
OCEANIA      14.0  47.785714  28.358091  13.0  23.25  45.0  70.0   88.0

Los porcentajes de población urbana son mayores en Europa con al menos el 75% de los paises por encima del 58,50 % de población urbana. En America la situación es similar. En Oceanía y Africa, sin embargo, la menos el 50 % de los paises no llegan al 50% de población urbana.

## CONCLUSIONES

Teniendo en cuenta lo analizado se puede concluir que el continente donde hay una mayor calidad de vida es Europa con valores para las variables analizadas bastante uniformes entre países lo que indica que la situación es semejante en la mayoría de los países.
Por otro lado, el continente con una menor calidad de vida se puede concluir que es Africa con valores para las variables analizadas muy diferentes en algunas de las variables lo que indica diferencias notables entre países. 
En Oceanía los valores obtenidos la sitúan en el rango bajo en cunato a calidad de vida aunque hay que tener en cuenta que los paises analizados han sido pocos por lo que un valor inusual en alguno de ellos influye sobremanera en el global del continente.
En Asia y America la calidad de vida es semejante aunque quizás pueda ser algo mejor en America. En estos continentes la variabilidad de los valores observados se puede considerar también bastante alta por lo que indica que las diferencias entre los paises obervados es significativa.